File: lib/domain/crop/dtos/crop_delete_usage.rb
# frozen_string_literal: true

module Domain
  module Crop
    module Dtos
      class CropDeleteUsage
        attr_reader :cultivation_plan_crops_count, :free_crop_plans_count, :pesticides_count

        def initialize(cultivation_plan_crops_count:, free_crop_plans_count:, pesticides_count:)
          @cultivation_plan_crops_count = cultivation_plan_crops_count
          @free_crop_plans_count = free_crop_plans_count
          @pesticides_count = pesticides_count
        end
      end
    end
  end
end

File: crates/agrr-domain/src/crop/dtos/crop_delete_usage.rs
/// Ruby: `Domain::Crop::Dtos::CropDeleteUsage`
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct CropDeleteUsage {
    pub cultivation_plan_crops_count: i32,
    pub free_crop_plans_count: i32,
    pub pesticides_count: i32,
}

impl CropDeleteUsage {
    pub fn new(cultivation_plan_crops_count: i32, free_crop_plans_count: i32, pesticides_count: i32) -> Self {
        Self { cultivation_plan_crops_count, free_crop_plans_count, pesticides_count }
    }
}
